### PR TITLE
ddl: use old name in changing column to refine error message

### LINKED ISSUE #25287: Change column type from `bit` to `varchar` reports 'incorrect utf8 value'
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
drop table if exists tbl_12;
create table tbl_12 (col_68 bit(45));
insert  into tbl_12 values (1174717);
alter table tbl_12 modify column col_68 varchar(31) collate utf8_unicode_ci ;
```

### 2. What did you expect to see? (Required)

MySQL:
```sql
Query OK, 1 row affected, 1 warning (0.06 sec)
Records: 1  Duplicates: 0  Warnings: 1
```

### 3. What did you see instead (Required)

TiDB
```sql
ERROR 1366 (HY000): incorrect utf8 value 00000011ecbd(
```

Error from log:
```
Error 1366: incorrect utf8 value 00000011ecbd(\u0000\u0000\u0000\u0011��) for column _Col$_col_137_0
```

### 4. What is your TiDB version? (Required)

master
```
commit f81ef5579551a0523d18b049eb25ab3375bcfb48 (HEAD, upstream/master)
Author: Yuanjia Zhang <zhangyuanjia@pingcap.com>
Date:   Tue Jun 8 20:08:28 2021 +0800

    planner: solve an incompatible issue about dynamic-mode and TiFlash and enable dynamic mode by default  (#25255)
```

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close #25287 

Problem Summary: The error message is confusing: 

```
ERROR 1366 (HY000): Incorrect string value '\xEC\xBD' for column '_Col$_a_0'
```

### What is changed and how it works?

What's Changed: use the old column name in the changing column.

### Check List 

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
  ```
  drop table if exists t;
  create table t (a bit(45));
  insert into t values (1174717);
  alter table t modify column a varchar(31) collate utf8mb4_general_ci;
  
  > ERROR 1366 (HY000): Incorrect string value '\xEC\xBD' for column 'a'
  ```
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by xiongjiwei** [comment]
why use manual test instead of UT

**Review by xiongjiwei — APPROVED** [review]

File: ddl/column_type_change_test.go [hunk 1]
```diff
@@ -2192,3 +2192,15 @@ func (s *testColumnTypeChangeSuite) TestChangeFromUnsignedIntToTime(c *C) {
 	tk.MustQuery("select a from t;").Check(testkit.Rows("18:08:57"))
 	tk.MustExec("drop table if exists t;")
 }
+
+// https://github.com/pingcap/tidb/issues/25287.
```
Comment (zimulala) [inline ddl/column_type_change_test.go:2196]:
```suggestion
// See https://github.com/pingcap/tidb/issues/25287.
```

**Review by zimulala — COMMENTED** [review]
Rest LGTM

**Review by zimulala — APPROVED** [review]

**Pull request merged**